requirements/lint-requirements.txt

```diff
@@ -2,6 +2,6 @@ ruff==0.6.4
 rstcheck==6.1.1
 black[jupyter]==23.7.0
 blacken-docs==1.18.0
-pre-commit==2.20.0
+pre-commit==4.0.1
```

Comment:
@JohannesJungbluth Thanks for the PR! I pinned `pre-commit`. We don't want `pre-commit` to slow down development (imagine pre-commit 5.0 is released with breaking changes). There is no strong reason to use the latest version of pre-commit.